### ISSUE TITLE
fails to process raw pld when the sensors are off for some yos

### ISSUE THREAD

**Issue by richardsc** [jklymak/pyglider#8]
Hi Jody -- this is a bit complicated, and might be better for a f2f discussion (if you have a bit of time), but I've successfully processed the real-time data from one of our missions (i.e. the `gli` and `pld` "sub" files), and am now trying to do the processing for the full "raw" dataset.

Because of battery limitations, we change the sampling of our gliders during the mission -- sometimes sampling only every second yo (both up and down, so 50% duty cycle), sometimes only sampling the up (50% duty cycle), and sometimes only sampling the up for every second yo (25% duty cycle).

I'm finding that in the `seaexplorer.merge_rawnc(rawncdir, rawncdir, deploymentyaml, kind='raw')` stage of the processing that the script fails, and I think it's because the `xarray` `open_mfdataset()` chokes when it finds a netcdf file with a size zero dimension:

Maybe this isn't something you've encountered before (I suspect the processing is a lot smoother when the CTD is sampling all the time), but I'd appreciate any insight or suggestions about how to handle this case.

I can make my processing script and data etc available in a self-contained reproducible example if you have time to take a look.

<details>

![image](https://user-images.githubusercontent.com/233584/117354981-60568400-ae88-11eb-82fb-7173cc02131e.png)

And see the following where you can see the first file that is causing the error:
```
raw_rawnc/sea021.0059.pld1.raw.0137_gpctd.nc
<xarray.Dataset>
Dimensions:             (time: 0)
Coordinates:
  * time                (time) float64 
Data variables:
    NAV_RESOURCE        (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    NAV_LONGITUDE       (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    NAV_LATITUDE        (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    NAV_DEPTH           (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    GPCTD_CONDUCTIVITY  (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    GPCTD_TEMPERATURE   (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    GPCTD_PRESSURE      (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    GPCTD_DOF           (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    Unnamed: 15         (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
    fnum                (time) float64 dask.array<chunksize=(0,), meta=np.ndarray>
```

**Comment by jklymak** [comment]
The merge step has been troublesome, and its not too surprising xarray chokes on this.  I *guess* the proper solution is not to *write* the empty netcdf files in the first place, rather than trying to make the merge work.  We could also report to xarray, however, as they should be able to gracefully handle empty netcdfs...

**Comment by jklymak** [comment]
i.e. around https://github.com/jklymak/pyglider/blob/1805b45bb72b0551b9b476f42d5f065cf6b0cf74/pyglider/seaexplorer.py#L146 (and the other writes) put a check and just don't write the file if empty...

**Comment by richardsc** [comment]
Thanks -- I'll try modifying to check for empty arrays. Want me to PR it if I figure it out?

I'm not familiar enough with xarray to be comfortable reporting this, so I'll leave that decision up to you 😄 .

**Comment by jklymak** [comment]
Was this closed by #10?

**Comment by richardsc** [comment]
Yes!
